pagerduty/resource_pagerduty_schedule.go

```diff
@@ -360,11 +361,19 @@ func expandScheduleLayers(v interface{}) ([]*pagerduty.ScheduleLayer, error) {
 			return nil, err
 		}
 
+		// If End is null, it means the layer does not end. The type of layer.*.end is schema.TypeString.
+		// A client should send a payload including `"end": null` to unset the end of layer.
+		var end *string
+		if rsl["end"].(string) != schema.TypeString.Zero().(string) {
+			e := rsl["end"].(string)
+			end = &e
+		}
```

Comment:
It might be better to add a generic helper function for this conversion so that other resources could also use it in the future for similar occasions.

in [util.go](https://github.com/PagerDuty/terraform-provider-pagerduty/blob/master/pagerduty/util.go)

```go
// NullString is a helper that returns a pointer to the string value passed in
// or nil if the string is empty.
func NullString(v string) *string {
	if v == "" {
		return nil
	}
	return &v
}
```

Reply:
@smaeda-ks Thanks for your suggestion! It was unclear why `rsl["end"].(string)` returns an empty value because `rsl` is `map[string]interface{}` while reading the code. Therefore, I used `schema.TypeString.Zero()` explicitly in e1c0f5d. 

Do you expect the util function is used when converting a resource of TypeString into a string pointer? If so, what do you think about defining the following function in utill.go?

```golang
func typeStringToStringPointer(v interface{}) *string {
	s := v.(string)
	if s == schema.TypeString.Zero().(string) {
		return nil
	}
	return &s
}
```

Reply:
While I totally understand your intention there (👍), I think it's somewhat fair to assume that TypeString schema values are safe to cast to string in that way from consistency and common-pattern perspective in provider development in general. Therefore, I wouldn't explicitly use `schema.TypeString.Zero()` in that way. It's also worth mentioning that there's a new [Terraform Plugin Framework](https://www.terraform.io/plugin/framework) implementation that may eventually replace the current SDKv2, which will change schema definitions as well. That said, hard-coding it might not be ideal for future migration.

So, I would stick with `v string` rather than `v interface{}` for this function argument, and pass `rsl["end"].(string)` instead if that makes sense. But that's just my two cents.

Reply:
That makes sense. I created a util function for the type conversion in c257825. Apart from that, I didn't know the new framework. that's good to know.  Thanks!